cmd/snap-confine/seccomp-support.c

```diff
@@ -109,12 +136,73 @@ static void validate_bpfpath_is_safe(const char *path)
 	}
 }
 
+static void sc_cleanup_sock_fprog(struct sock_fprog *prog)
+{
+	free(prog->filter);
+	prog->filter = NULL;
+}
+
+static void sc_must_read_filter_from_file(FILE *file, uint32_t len_bytes, char *what, struct sock_fprog *prog)
+{
+	prog->len = len_bytes / sizeof(struct sock_filter);
+	prog->filter = (struct sock_filter *)malloc(len_bytes);
```

Comment:
Can `len_bytes` be 0? In that case, glibc will handle it, but will technically allocate 1 byte. Maybe we could use NULL instead in that case.